src/query/interpreter.cpp

```diff
@@ -1236,14 +1251,29 @@ Interpreter::Interpreter(InterpreterContext *interpreter_context) : interpreter_
   MG_ASSERT(interpreter_context_, "Interpreter context must not be NULL");
 }
 
-PreparedQuery Interpreter::PrepareTransactionQuery(std::string_view query_upper,
-                                                   const std::map<std::string, storage::PropertyValue> &metadata) {
+auto DetermineTxTimeout(std::optional<int64_t> tx_timeout, InterpreterConfig const &config) -> TxTimeout {
```

Comment:
What is the reason for using `auto` + trailing return type? Shouldn't we use that only in some instances of lamdas usage?